model/flow/sealing_segment.go

```diff
@@ -63,6 +63,9 @@ type SealingSegment struct {
 	// sealed state, when the first block contains no seal.
 	// If the first block in the segment contains seal, then this field is `nil`
 	FirstSeal *Seal
+
+	// Seal is the seal for the first block (the lowest height) in the segment
+	Seal *Seal
```

Comment:
I had to add Seal to sealing segment, because we want to make sure the root block 's seal is also indexed. Since the root block' seal is created by a sealing segment, I had to ensure the sealing segment's first block is sealed, and has the seal. This can be guaranteed, since the `validateSegment` call will look for the seal. So once `validateSegment` finds the seal for the first block, we just add it here to guarantee a SealingSegment always have a Seal.

Reply:
I don't understand why this is necessary. To the best of my understanding, 'FirstSeal' should be sufficient. 
In your later comment, you argue that a sealing segment can only seal one block; I thing the seal for this one block is 'FirstSeal' right (?)

Reply:
I don't think the new `Seal` field is necessary either. We already are guaranteed to have the seal for the lowest block in the sealing segment:
* If we have a root sealing segment, then `FirstSeal` is set and it is the seal for the lowest (only) block in the sealing segment by definition
* If we have a non-root sealing segment, then by definition, the seal for the lowest block must be included in the highest block (or its ancestor within the sealing segment)